webapp/src/selectors.ts

```diff
@@ -35,199 +44,243 @@ import {pluginId} from './manifest';
+const reactionsInCalls = (state: GlobalState): usersReactionsState => {
+    return pluginState(state).reactions;
 };
 
-export const voiceReactions = (state: GlobalState): Reaction[] => {
-    return pluginState(state).reactionStatus[connectedChannelID(state) || '']?.reactions || [];
-};
-
-export const voiceUsersStatusesInChannel = (state: GlobalState, channelID: string) => {
-    return pluginState(state).voiceUsersStatuses[channelID] || {};
-};
+export const reactionsInCurrentCall: (state: GlobalState) => Reaction[] =
+    createSelector(
+        'reactionsInCurrentCall',
+        reactionsInCalls,
+        channelIDForCurrentCall,
+        (reactions, channelID) => reactions[channelID]?.reactions || [],
+    );
 
-export const voiceChannelCallStartAt = (state: GlobalState, channelID: string): number | undefined => {
-    return pluginState(state).voiceChannelCalls[channelID]?.startAt;
+export const callStartAtForCallInChannel = (state: GlobalState, channelID: string): number => {
+    return pluginState(state).calls[channelID]?.startAt || 0;
 };
 
-export const voiceChannelCalls = (state: GlobalState): { [channelID: string]: callState} =>
-    pluginState(state).voiceChannelCalls;
+export const callStartAtForCurrentCall: (state: GlobalState) => number =
+    createSelector(
+        'callStartAtForCurrentCall',
+        calls,
+        channelIDForCurrentCall,
+        (callsStates, channelID) => callsStates[channelID]?.startAt ||
+          window.callsClient?.initTime ||
+          window.opener.callsClient?.initTime || 0,
+    );
 
-export const voiceChannelCallInCurrentChannel: (state: GlobalState) => callState =
+export const callInCurrentChannel: (state: GlobalState) => callState =
     createSelector(
-        'callStartAtInCurrentChannel',
-        voiceChannelCalls,
+        'callInCurrentChannel',
+        calls,
         getCurrentChannelId,
-        (calls, currChannelId) => calls[currChannelId],
+        (callsStates, currChannelId) => callsStates[currChannelId],
     );
 
-export const voiceChannelCallID = (state: GlobalState, channelID: string): string | undefined => {
-    return pluginState(state).voiceChannelCalls[channelID]?.ID;
+export const idForCallInChannel = (state: GlobalState, channelID: string): string | undefined => {
+    return pluginState(state).calls[channelID]?.ID;
 };
 
-export const voiceChannelCallOwnerID = (state: GlobalState, channelID: string): string | undefined => {
-    return pluginState(state).voiceChannelCalls[channelID]?.ownerID;
+export const callOwnerIDForCallInChannel = (state: GlobalState, channelID: string): string | undefined => {
+    return pluginState(state).calls[channelID]?.ownerID;
 };
 
-export const voiceChannelCallHostID = (state: GlobalState, channelID: string) => {
-    return pluginState(state).voiceChannelCalls[channelID]?.hostID;
+const hostsInCalls = (state: GlobalState): hostsState => {
```

Comment:
I'm not crazy about the name here... But I can't think of anything better. Something like `hostByCall`?

Reply:
It's really just internal given it's the base selector so we may as well just call it `hosts` or `callsHosts`. Ultimately it doesn't matter that much I guess.